app/src/main/java/com/github/se/cyrcle/ui/profile/DisplayProfileComponent.kt

```diff
@@ -67,21 +74,86 @@ fun DisplayProfileComponent(user: User?, extras: @Composable () -> Unit) {
 
               Spacer(modifier = Modifier.height(8.dp))
 
-              Text(
-                  text =
-                      stringResource(
-                          R.string.display_user_tag_format, range.symbol, level, username),
-                  style = MaterialTheme.typography.bodyMedium,
-                  color =
-                      if (range.color == stringResource(R.string.rainbow_text_color)) {
-                        MaterialTheme.colorScheme.onSurface // TODO rainbow color
-                      } else {
-                        Color(parseColor(range.color))
-                      },
-                  modifier = Modifier.testTag("DisplayUsernameWithLevel"))
+              UserTagText(
+                  ownerUsername = username,
+                  symbol = range.symbol,
+                  color = range.color,
+                  level = level,
+                  modifier = Modifier.testTag("DisplayUsernameWithLevel"),
+                  style = MaterialTheme.typography.bodyMedium)
 
               Spacer(modifier = Modifier.height(16.dp))
             }
         extras()
       }
 }
+
+@Composable
+fun RainbowText(
+    text: String,
+    modifier: Modifier = Modifier,
+    style: TextStyle = MaterialTheme.typography.bodySmall,
+    maxLines: Int = Int.MAX_VALUE,
+    overflow: TextOverflow = TextOverflow.Clip
+) {
+  val Orange = Color(0xFFFFA500) // Hex for orange
+  val Indigo = Color(0xFF4B0082) // Hex for indigo
+  val Violet = Color(0xFF8A2BE2) // Hex for violet
+
+  Text(
+      text = text,
+      modifier = modifier,
+      style =
+          style.copy(
+              brush =
+                  Brush.horizontalGradient(
+                      colors =
+                          listOf(
+                              Color.Red,
+                              Orange,
+                              Color.Yellow,
+                              Color.Green,
+                              Color.Blue,
+                              Indigo,
+                              Violet))),
+      maxLines = maxLines,
+      overflow = overflow)
+}
+
+@Composable
+fun UserTagText(
+    ownerUsername: String,
+    symbol: String,
+    color: String,
+    level: Int,
+    modifier: Modifier = Modifier,
+    style: TextStyle = MaterialTheme.typography.bodyMedium,
+    maxLines: Int = Int.MAX_VALUE,
+    overflow: TextOverflow = TextOverflow.Clip
+) {
+  val formattedText = stringResource(R.string.display_user_tag_format, symbol, level, ownerUsername)
+
+  if (color == stringResource(R.string.rainbow_text_color)) {
+    RainbowText(
+        text = formattedText,
+        modifier = modifier,
+        style =
+            style.copy(
+                fontWeight = FontWeight.Bold,
+                shadow =
+                    Shadow(
```

Comment:
This is nice. The text is much more readable!! I think level 100 users will be happy to pop so much!